core/action/service.go

```diff
@@ -19,11 +40,22 @@ func (s Service) Get(ctx context.Context, id string) (Action, error) {
 }
 
 func (s Service) Create(ctx context.Context, action Action) (Action, error) {
+	currentUser, err := s.userService.FetchCurrentUser(ctx)
+	if err != nil {
+		return Action{}, fmt.Errorf("%w: %s", user.ErrInvalidEmail, err.Error())
+	}
+
 	newAction, err := s.repository.Create(ctx, action)
 	if err != nil {
 		return Action{}, err
 	}
 
+	logData := newAction.ToActionLogData()
+	if err := s.activityService.Log(ctx, AuditKeyActionCreate, currentUser.ID, logData); err != nil {
+		logger := grpczap.Extract(ctx)
```

Comment:
Rather than fetching the logger from the contex, would be better to attach the logger in the service struct domain 

```
type Service struct {
  logger *zap.Logger
}
```

So calling the logger would be `s.logger.Error`